File: Práctica 1/S1/src/s1/carreraCarretera.java
package s1;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Random;

public class carreraCarretera extends Carrera {
    
    carreraCarretera(int codigo){
        super(codigo);
    }
    
    
    
    public  void run(){
        System.out.println("Comienza la carrera de Carretera");
        Collections.shuffle(this.bicicletas); //Desordenamos los participantes, para simular que están compitiendo
        
        int abandonos = (int) ((int)this.bicicletas.size() * 0.1);
        
        System.out.println("Se han retirado " + abandonos + " ciclistas.");
        
        for(int i=0;i<abandonos;i++){
            Random rand = new Random();
            int caido = rand.nextInt(this.bicicletas.size());
            System.out.println("El participante " + this.bicicletas.get(caido).getID() + " se ha retirado");
            this.bicicletas.remove(caido);
        }
        
    }
    
}

File: Práctica 1/S1R/lib/CarreraCarretera.rb
require('./Ruby/Carrera.rb');

module Ruby
	public
	class CarreraCarretera < Carrera

		public
		def clonar()
			return self
		end

		public
		def initialize(numero)
			super(numero)
		end
    
    public 
    def correr
      puts "Comienza la carrera de carretera"
      @bicis = @bicis.shuffle
      tamanio=@bicis.lenght
      abandonos = tamanio*0.1
      puts "El numero de abandonos es #{abandonos}"
      for i in (0..abandonos)
        caido=rand(1..tamanio)
        puts "El participante #{caido} se ha retirado de la carrera de carretera"
        @bicis.delete(caido)
      end
      
    end
    
	end
end
